test/integration/puppetlabs/services/puppet_admin/puppet_admin_int_test.clj

```diff
@@ -178,5 +181,26 @@
            (is (= 204 (:status response))
                (ks/pprint-to-string response))))))))
 
+(deftest ^:integration admin-api-pool-timeout-test
+  (testing "pool lock timeout results in 503"
+    (logutils/with-test-logging
+      (bootstrap/with-puppetserver-running
+       app
+       {:jruby-puppet {:flush-timeout 0}}
+       ; Borrow a jruby instance so that the timeout of 0 is reached immediately
+       (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
+             pool-context (jruby-protocol/get-pool-context jruby-service)
+             pool (jruby-core/get-pool pool-context)
+             borrowed-instance (.borrowItem pool)]
+         (try
+           (let [response (http-client/delete
+                           (str "https://localhost:8140/puppet-admin-api/v1/jruby-pool")
+                           ssl-request-options)]
+             (is (= 503 (:status response)))
+             (is (= "Timeout limit reached before lock could be granted"
```

Comment:
I think this is going to need to be changed once you update to a released jruby-utils, cuz the string there has changed.... But you'll probably realize that cuz the test will fail :P.